socket.go

```diff
@@ -223,7 +225,7 @@ func (socket *mongoSocket) ServerInfo() *mongoServerInfo {
 // InitialAcquire obtains the first reference to the socket, either
 // right after the connection is made or once a recycled socket is
 // being put back in use.
-func (socket *mongoSocket) InitialAcquire(serverInfo *mongoServerInfo, timeout time.Duration) error {
+func (socket *mongoSocket) InitialAcquire(serverInfo *mongoServerInfo, dialInfo *DialInfo) error {
```

Comment:
The  `*mongoSocket` already contains dialInfo at the time of this call as far as I can tell (maybe there is another place where it doesn't (?))

Reply:
It can be called from a sync task that can have a different timeout configured, so it respects the value it was provided rather than the socket timeout.